tests-invoke

```diff
@@ -252,8 +256,21 @@ class PullTask(object):
             cmd = ["timeout", "120m", entry_point]
             if opts.verbose:
                 cmd.append("--verbose")
-            ret = subprocess.call(cmd)
-
+            # Popen instead of call; while not etc... (SubProcess.poll)
+            if ret:
+                sys.stderr.write('Collision detected: {0}\n'.format(ret))
+                return None
+            p = subprocess.Popen(cmd)
+            while p.poll() is None:
+                ret = self.detect_collisions(opts, True)
+                if ret:
+                    sys.stderr.write('Collision detected: {0}\n'.format(ret))
+                    ret = None
+                    p.terminate()
```

Comment:
Do a `break` here, or do the sleeping in `else:`. No need to sleep another minute after cancelling.

Also, does the child get reaped automatically, or do we need a `p.wait()` here?